### PR TITLE
perm_long_chain_assignment

### PR DESCRIPTION
**Pull request by snuffysasa**
Create a long chain assignment out of 3 or more consecutive assignments with equal rvalue

### CODE REVIEW

**Comment by snuffysasa** [comment]
Okay I know the nested while loops in the rec(block) function is quite C-style and not very pythonic.

maybe you can help me out?

File: src/randomizer.py [hunk 1]
```diff
@@ -1992,6 +1992,65 @@ def rec(block: Block) -> None:
...
+    statements = ast_util.get_block_stmts(block, True)
+
+    # Merge all statements into long chain assignment at start_idx
+    stmt = statements[start_idx]
+    assert isinstance(stmt, ca.Assignment)
+    for i in range(start_idx + 1, end_idx + 1):
+        additional_stmt = statements[i]
+        assert isinstance(additional_stmt, ca.Assignment)
+        new_lvalue = additional_stmt.lvalue
+        stmt.rvalue = ca.Assignment("=", stmt.lvalue, stmt.rvalue)  # copy of stmt
```
Comment (simonlindholm) [inline src/randomizer.py:2046]:
```suggestion
        stmt.rvalue = ca.Assignment("=", stmt.lvalue, copy.deepcopy(stmt.rvalue))
```

**Review by simonlindholm — COMMENTED** [review]
Looks good, I can see why you want this separated from perm_chain_assignment (assignments being consecutive, no logic for `a = b; c = a`). I think the nested whiles look very sensible

File: default_weights.toml [hunk 2]
```diff
@@ -29,13 +29,15 @@ perm_compound_assignment = 5
 perm_remove_ast = 5
 perm_duplicate_assignment = 5
 perm_chain_assignment = 5
+perm_long_chain_assignment = 1
```
[the remark this answers is not in the document]
Comment (simonlindholm) [reply default_weights.toml:32]: That's fair! Maybe it should be lower for mwcc too then...?

File: src/randomizer.py (hunk 1 above)
Comment (snuffysasa) [reply src/randomizer.py:2046]: wait, can't I just make this whole thing:   
`stmt.rvalue = copy.deepcopy(stmt)`

**Pull request merged**